plugins/star-rating/frontend/public/javascripts/countly.views.js

```diff
@@ -503,6 +547,75 @@ var WidgetDetail = countlyVue.views.create({
         backToWidgets: function() {
             window.location.hash = '#/' + countlyCommon.ACTIVE_APP_ID + '/feedback/ratings/widgets';
         },
+        calCumulativeData: function() {
+            var self = this;
+            // reset values
+            self.count = 0;
+            self.avg = 0;
+            self.sum = 0;
+            // reset cumulative data
+            self.cumulativeData = [{
+                count: 0,
+                percent: 0
+            }, {
+                count: 0,
+                percent: 0
+            }, {
+                count: 0,
+                percent: 0
+            }, {
+                count: 0,
+                percent: 0
+            }, {
+                count: 0,
+                percent: 0
+            }];
+
+            var ratingArray = [];
+            var result = self.rating;
+            var periodArray = countlyCommon.getPeriodObj().currentPeriodArr;
+
+            // prepare cumulative data by period object
+            for (var i = 0; i < periodArray.length; i++) {
+                var dateArray = periodArray[i].split('.');
+                var year = dateArray[0];
+                var month = dateArray[1];
+                var day = dateArray[2];
+                if (result[year] && result[year][month] && result[year][month][day]) {
+                    for (var rating in result[year][month][day]) {
+                        if (self.matchPlatformVersion(rating)) {
+                            var rank = (rating.split("**"))[2];
+                            if (self.cumulativeData[rank - 1]) {
+                                self.cumulativeData[rank - 1].count += result[year][month][day][rating].c;
+                                self.count += result[year][month][day][rating].c;
+                                self.sum += (result[year][month][day][rating].c * rank);
+                                self.avg = self.sum / self.count;
+                                var times = result[year][month][day][rating].c;
+                                while (times--) {
```

Comment:
`infinite loop gods` liked this.

so yeah, this one seemed a bit dangerous to me 😄 what if `times` doesn't reach to 0?